framework/src/mesh/MooseMesh.C

```diff
@@ -401,6 +406,26 @@ MooseMesh::prepare(const MeshBase * const mesh_to_clone)
   {
     const auto add_subdomain_id = getParam<std::vector<SubdomainID>>("add_subdomain_ids");
     _mesh_subdomains.insert(add_subdomain_id.begin(), add_subdomain_id.end());
+
+    // do we have subdomain names explicitly given?
+    if (isParamValid("add_subdomain_names"))
+    {
+      const auto add_subdomain_name = getParam<std::vector<SubdomainName>>("add_subdomain_names");
+      const int n_ids = add_subdomain_id.size();
+      const int n_names = add_subdomain_name.size();
+
```

Comment:
Use `getParam<SubdomainID, SubdomainName>("add_subdomain_ids", "add_subdomain_names")` to get a vector of pairs with automatic checking for equal number of entries.

Reply:
Dear @dschwen, This looks elegant! 

But then the user *must* provide both, `add_subdomain_ids` and `add_subdomain_names`? Wouldn't that break all existing models that only provide `add_subdomain_ids` to date (and not `add_subdomain_names`)?

Or to put it another way: Will Moose demand in future that both `add_subdomain_names` and `add_subdomain_names` must be given?

Reply:
Not if you keep the `isParamValid("add_subdomain_names")` and conditionally fetch the vector of pairs